### ISSUE TITLE
Cannot install using Composer in Magento 2.1.17

### ISSUE THREAD

**Issue by insaneCreations** [tig-nl/postnl-magento2#103]
composer require tig/postnl-magento2
Using version ^1.7 for tig/postnl-magento2
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Installation request for zendframework/zend-servicemanager (locked at 2.5.1) -> satisfiable by zendframework/zend-servicemanager[2.5.1].
    - Installation request for tig/postnl-magento2 ^1.7 -> satisfiable by tig/postnl-magento2[1.7.0, 1.7.1, 1.7.2, 1.7.3, 1.7.4].
    - Installation request for magento/framework (locked at 100.1.17) -> satisfiable by magento/framework[100.1.17].
    - Can only install one of: colinmollenhour/php-redis-session-abstract[v1.2.2, v1.3.4].
    - Can only install one of: colinmollenhour/php-redis-session-abstract[v1.3.4, v1.2.2].
    - Can only install one of: colinmollenhour/php-redis-session-abstract[v1.2.2, v1.3.4].
    - Conclusion: install colinmollenhour/php-redis-session-abstract v1.2.2
    - Installation request for colinmollenhour/php-redis-session-abstract (locked at v1.3.4) -> satisfiable by colinmollenhour/php-redis-session-abstract[v1.3.4].

**Comment by tig-daanvandenbergh** [comment]
Hi @insaneCreations,

I've adjusted the title to represent the issue more accurately. 

The following line reveals the issue:
`Can only install one of: colinmollenhour/php-redis-session-abstract[v1.2.2, v1.3.4].`

The problem isn't `tig/postnl-magento2`, but Collin Mollenhour's extension. While trying to install the PostNL-module, composer runs its routine check and updates other modules within the project. 

Like it says in the log, `colinmollenhour/php-redis-session-abstract` is locked at v1.3.4, while another extension (possibly Magento itself, is trying to install v1.2.2). Check your webroot's`composer.json`-file to see if this extension is explicitly listed there and if so, remove it (because you shouldn't require it in Magento 2's composer.json-file).

After that try removing your `composer.lock`-file and running `composer require tig/postnl-magento2 --update-with-dependencies`.

I'm closing this issue, since it's not PostNL related.

**Comment by greatwilliamworld** [comment]
Hi, tig-daanvandenbergh.
I met a same issue with another module, but your solution did not work.
*********************************
Using version ^3.1 for cybersource/global-payment-management
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Installation request for cybersource/global-payment-management ^3.1 -> satisfiable by cybersource/global-payment-management[3.1.0, 3.1.1, 3.1.2].
    - Installation request for magento/framework (installed at 100.1.18) -> satisfiable by magento/framework[100.1.18].
    - Can only install one of: colinmollenhour/php-redis-session-abstract[v1.2.2, v1.3.4].
    - Can only install one of: colinmollenhour/php-redis-session-abstract[v1.3.4, v1.2.2].
    - Conclusion: install colinmollenhour/php-redis-session-abstract v1.2.2
    - Installation request for colinmollenhour/php-redis-session-abstract (installed at v1.3.4) -> satisfiable by colinmollenhour/php-redis-session-abstract[v1.3.4].

Installation failed, reverting ./composer.json to its original content.
*********************************

Could you please let me know better solution?

**Comment by tig-daanvandenbergh** [comment]
Hi @greatwilliamworld,

Thanks for your question on our support forum.

What I can see from the error, is that another module must be requiring a different (older) version of `colinmollenhour/php-redis-session-abstract`. Try updating that module to a higher version, requiring version v1.3.4. 

Good luck!

**Comment by greatwilliamworld** [comment]
Thank you for your quick reply.
My site is Magento 2.1.18 and module version is latest one which was provided as composer.json from provider. I have tried to find it, but no success... let me try again

**Comment by greatwilliamworld** [comment]
I searched it using grep -ir colinmollenhour/php-redis-session-abstract commands on whole server, but I couldn't find any module which needed 1.2.2 version.

Could you please solution for me?
Many thanks.

**Comment by tig-daanvandenbergh** [comment]
Hi @greatwilliamworld,

Perhaps you should reach out to the developers of `cybersource/global-payment-management`. They might be able to help you.

**Comment by greatwilliamworld** [comment]
Their is no any reply from them.
Anyway, thank you!
